config/remote/analytics.go

```diff
@@ -1,168 +1,61 @@
 package remote
 
 import (
-	"encoding/json"
-	"fmt"
-	"log"
-
 	"github.com/NordSecurity/nordvpn-linux/events"
-	"github.com/NordSecurity/nordvpn-linux/internal"
 )
 
-type UserInfo struct {
-	AppVersion   string
-	Country      string
-	ISP          string
-	RolloutGroup int
-}
-
-// EventDetails holds optional details for an event.
-type EventDetails struct {
-	Error   string `json:"error"`
-	Message string `json:"message"`
-}
-
-// Event is the main analytics event structure.
-type Event struct {
-	UserInfo `json:"-"`
-	EventDetails
-	MessageNamespace string `json:"namespace"`
-	Subscope         string `json:"subscope"`
-	Client           string `json:"client"`
-	Event            string `json:"event"`
-	Result           string `json:"result"`
-	FeatureName      string `json:"-"`
-	RolloutPerformed bool   `json:"-"`
-}
-
-func NewDownloadSuccessEvent(info UserInfo, client string, featureName FeatureName) Event {
-	return Event{
-		UserInfo:    info,
-		Client:      client,
-		FeatureName: featureName.String(),
-		Event:       DownloadSuccess.String(),
-	}
-}
-
-func NewDownloadFailureEvent(info UserInfo, client string, featureName FeatureName, errorKind DownloadErrorKind, errorMessage string) Event {
-	return Event{
-		UserInfo:    info,
-		Client:      client,
-		FeatureName: featureName.String(),
-		Event:       DownloadFailure.String(),
-		EventDetails: EventDetails{
-			Error:   errorKind.String(),
-			Message: errorMessage,
+// Analytics defines an interface for reporting various analytics events related to downloads,
+// local usage, JSON parsing, and partial rollouts. Implementations of this interface are
+// responsible for handling the notification logic for each event type.
+type Analytics interface {
+	NotifyDownload(string, string)
+	NotifyDownloadFailure(string, string, DownloadError)
```

Comment:
Also, maybe we could accept whole structs here instead of separate args? With structs it would be harder to put args in wrong order if we have multiple args with the same type

The implementations also use factory methods to produce event and immediately transform it to `DebuggerEvent` like:

```
func (ma *MooseAnalytics) NotifyDownload(client, featureName string) {
	ma.publisher.Publish(*NewDownloadSuccessEvent(ma.ctx.UserInfo, client, featureName).ToDebuggerEvent())
}
```
So maybe we could get rid of one layer this way.

The downside is that you would have to construct the event at the moment of calling the methods, but that's what we have so far in `main` - we construct the event and call `.Publish(event)`.

Not a big deal, just something to take into account